geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geotools/SftArgResolver.scala

```diff
@@ -22,84 +23,100 @@ import scala.util.{Failure, Success, Try}
  * including sft strings (e.g. name:String,age:Integer,*geom:Point)
  * and typesafe config.
  */
-object SftArgResolver extends LazyLogging {
-
-  // Important to setAllowMissing to false bc else you'll get a config but it will be empty
-  val parseOpts =
-    ConfigParseOptions.defaults()
-      .setAllowMissing(false)
-      .setClassLoader(null)
-      .setIncluder(null)
-      .setOriginDescription(null)
-      .setSyntax(null)
-
-  /**
-   * @return the SFT parsed from the Args
-   */
-  def getSft(specArg: String, featureName: String = null): Option[SimpleFeatureType] =
-    getLoadedSft(specArg, featureName)
-      .orElse(parseSpecString(specArg, featureName))
-      .orElse(parseConfStr(specArg, featureName))
-      .orElse(parseSpecStringFile(specArg, featureName))
-      .orElse(parseConfFile(specArg, featureName))
+object SftArgResolver extends AbstractArgResolver[SimpleFeatureType, (String, String)] with LazyLogging {
+
+  import ArgTypes._
+
+  override def argType(args: argTuple) = {
```

Comment:
similarly, if you use a case class you can include that in the method definitions to make it much clearer:
`override def argType(args: SpecAndName)`